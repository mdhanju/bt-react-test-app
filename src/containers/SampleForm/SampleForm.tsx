import React from 'react'
import {
    Button,
    FormGroup,
    Label,
    Input,
    Card,
    CardBody,
    CardTitle,
    Col,
    Row,
    DropdownItem
} from 'reactstrap'

import { connect } from 'react-redux'
import { ThunkDispatch } from 'redux-thunk'
import { State } from '../../store/reducers/rootReducer'
import { submitForm } from '../../store/actions/applicationInitActions'
import { RegisterAction } from '../../store/actions/applicationInitActions'

import ContentHolder from '../../components/ContentHolder'

const SampleForm = (props?: any): React.ReactElement => {
    (window as any).btPageName = "BT React Test Register Page"
    return (
        <ContentHolder>
            <Card>
                <CardBody>
                    <CardTitle>
                        <h2>Register</h2>
                    </CardTitle>
                    <DropdownItem divider className="mb-3" />
                    <Row form>
                        <Col md={6}>
                            <FormGroup>
                                <Label for="email">Email</Label>
                                <Input
                                    type="email"
                                    placeholder="12345@example.com"
                                    id="email"
                                />
                            </FormGroup>
                        </Col>
                        <Col md={6}>
                            <FormGroup>
                                <Label for="password">Password</Label>
                                <Input type="password" id="password" />
                            </FormGroup>
                        </Col>
                    </Row>
                    <FormGroup>
                        <Label for="address">Address</Label>
                        <Input
                            type="text"
                            placeholder="1234 Main St"
                            id="address"
                        />
                    </FormGroup>
                    <FormGroup>
                        <Label for="address2">Address 2</Label>
                        <Input
                            type="text"
                            placeholder="Apartment, studio, or floor"
                            id="address2"
                        />
                    </FormGroup>

                    <Row form>
                        <Col md={6}>
                            <FormGroup>
                                <Label for="city">City</Label>
                                <Input type="text" id="city" />
                            </FormGroup>
                        </Col>
                        <Col md={4}>
                            <FormGroup>
                                <Label for="state">State</Label>
                                <Input type="text" id="state" />
                            </FormGroup>
                        </Col>
                        <Col md={2}>
                            <FormGroup>
                                <Label for="zip">Zip</Label>
                                <Input type="text" id="zip" />
                            </FormGroup>
                        </Col>

                        <Col md={12}>
                            <Button block color="primary" onClick={async () => {
                                try {
                                    await props.submitForm()
                                    props.history.push('/profile')
                                } catch (error) {
                                    console.log(error)
                                }
                            }}>
                Register
                            </Button>
                        </Col>
                    </Row>
                </CardBody>
            </Card>
        </ContentHolder>
    )
}

const mapStateToProps = (state: State): object => ({
    register: state.appContext.register,
})

const mapDispatchToProps = (
    dispatch: ThunkDispatch<State, void, RegisterAction>
): object => ({
    submitForm: (): Promise<void> => dispatch(submitForm())
})

const HomeWithRedux: /* eslint-disable-line @typescript-eslint/no-explicit-any */ any = connect(
    mapStateToProps,
    mapDispatchToProps
)(SampleForm)
export default HomeWithRedux
