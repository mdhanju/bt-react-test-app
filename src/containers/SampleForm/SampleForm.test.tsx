import React from 'react'

import SampleForm from './SampleForm'

describe('<SampleForm />', () => {
    it('should render properly without crashing', () => {
        const wrapper = shallow(<SampleForm />)
        expect(wrapper.exists()).toBe(true)
        expect(wrapper.getElements()).toMatchSnapshot()
    })
})
