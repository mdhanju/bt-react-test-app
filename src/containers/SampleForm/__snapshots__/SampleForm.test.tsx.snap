// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SampleForm /> should render properly without crashing 1`] = `
Array [
  <ContentHolder>
    <Card
      tag="div"
    >
      <CardBody
        tag="div"
      >
        <CardTitle
          tag="div"
        >
          <h2>
            Register
          </h2>
        </CardTitle>
        <DropdownItem
          className="mb-3"
          divider={true}
          tag="button"
          toggle={true}
        />
        <Row
          form={true}
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Col
            md={6}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <FormGroup
              tag="div"
            >
              <Label
                for="email"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                Email
              </Label>
              <Input
                id="email"
                placeholder="12345@example.com"
                type="email"
              />
            </FormGroup>
          </Col>
          <Col
            md={6}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <FormGroup
              tag="div"
            >
              <Label
                for="password"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                Password
              </Label>
              <Input
                id="password"
                type="password"
              />
            </FormGroup>
          </Col>
        </Row>
        <FormGroup
          tag="div"
        >
          <Label
            for="address"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Address
          </Label>
          <Input
            id="address"
            placeholder="1234 Main St"
            type="text"
          />
        </FormGroup>
        <FormGroup
          tag="div"
        >
          <Label
            for="address2"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Address 2
          </Label>
          <Input
            id="address2"
            placeholder="Apartment, studio, or floor"
            type="text"
          />
        </FormGroup>
        <Row
          form={true}
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Col
            md={6}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <FormGroup
              tag="div"
            >
              <Label
                for="city"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                City
              </Label>
              <Input
                id="city"
                type="text"
              />
            </FormGroup>
          </Col>
          <Col
            md={4}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <FormGroup
              tag="div"
            >
              <Label
                for="state"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                State
              </Label>
              <Input
                id="state"
                type="text"
              />
            </FormGroup>
          </Col>
          <Col
            md={2}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <FormGroup
              tag="div"
            >
              <Label
                for="zip"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                Zip
              </Label>
              <Input
                id="zip"
                type="text"
              />
            </FormGroup>
          </Col>
          <Col
            md={12}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Button
              block={true}
              color="primary"
              tag="button"
            >
              Register
            </Button>
          </Col>
        </Row>
      </CardBody>
    </Card>
  </ContentHolder>,
]
`;
